test/integration/040_init_test/test_init.py

```diff
@@ -26,18 +29,32 @@ def models(self):
         return "models"
 
     @use_profile('postgres')
-    def test_postgres_init_task(self):
-        project_name = self.get_project_name()
-        self.run_dbt(['init', project_name, '--adapter', 'postgres'])
-
-        assert os.path.exists(project_name)
-        project_file = os.path.join(project_name, 'dbt_project.yml')
-        assert os.path.exists(project_file)
-        with open(project_file) as fp:
-            project_data = yaml.safe_load(fp.read())
-
-        assert 'config-version' in project_data
-        assert project_data['config-version'] == 2
-
-        git_dir = os.path.join(project_name, '.git')
-        assert not os.path.exists(git_dir)
+    @mock.patch('click.confirm')
+    @mock.patch('click.prompt')
+    def test_postgres_init_task_in_project(self, mock_prompt, mock_confirm):
+        manager = Mock()
+        manager.attach_mock(mock_prompt, 'prompt')
+        manager.attach_mock(mock_confirm, 'confirm')
+        manager.confirm.side_effect = ["y"]
+        manager.prompt.side_effect = [
+            1,
+            4,
+            "localhost",
+            5432,
+            "test_user",
+            "test_password",
+            "test_db",
+            "test_schema",
+        ]
+        self.run_dbt(['init', '--profiles-dir', 'dbt-profile'], profiles_dir=False)
+        manager.assert_has_calls([
+            call.confirm('The profile test already exists in /Users/niall.woodward/.dbt/profiles.yml. Continue and overwrite it?'),
```

Comment:
I'm struggling to work out how to get the profile directory to be one that is consistent between testing environments. I think it relates to https://github.com/dbt-labs/dbt/pull/1527 - but would definitely appreciate it if one of the core maintainers could take a look as the test running code looks to be quite in-depth.